polaris-react/src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx

```diff
@@ -270,19 +277,21 @@ export class PopoverOverlay extends PureComponent<PopoverOverlayProps, State> {
     const target = event.target as HTMLElement;
     const {
       contentNode,
-      props: {activator, onClose},
+      props: {activator, onClose, hasPortalDescendants},
     } = this;
-    const isDescendant =
-      contentNode.current != null &&
-      nodeContainsDescendant(contentNode.current, target);
+    const wasDescendant = hasPortalDescendants
+      ? wasPolarisPortalDescendant(event.composedPath(), this.context.container)
+      : contentNode.current != null &&
```

Comment:
Might be worth creating a named function for the else condition, so it's obvious what this is intending to validate.

Reply:
Agree